apps/frontend/src/store/search.ts

```diff
@@ -0,0 +1,550 @@
+  @Action
+  addTitleSearch(title: string) {
+    if (this.searchTerm.trim() !== '') {
+      this.context.commit(
+        'SET_SEARCH',
+        `${this.searchTerm} AND title:${title}`
+      );
+    } else {
+      this.context.commit('SET_SEARCH', `title:${title}`);
+    }
+  }
+
+  /** Adds a title filter */
+  @Action
+  addTitleFilter(title: string | string[]) {
+    this.context.commit('ADD_TITLE', title);
+  }
+
+  @Mutation
+  ADD_TITLE(title: string | string[]) {
+    this.titleSearchTerms = this.titleSearchTerms.concat(title);
+  }
+
+  /** Sets the title filters */
+  @Mutation
+  SET_TITLE(titles: string[]) {
+    this.titleSearchTerms = titles;
+  }
+
+  /** Clears all title filters */
+  @Mutation
+  CLEAR_TITLE() {
+    this.titleSearchTerms = [];
+  }
+
+  // CCI ID Filtering
+
+  /** Adds or replaces severity in the search bar */
+  @Action
+  addCCIIdSearch(cciId: string) {
+    if (this.searchTerm.trim() !== '') {
+      this.context.commit('SET_SEARCH', `${this.searchTerm} AND nist:${cciId}`);
+    } else {
+      this.context.commit('SET_SEARCH', `nist:${cciId}`);
+    }
+  }
+
+  /** Adds CCI id to filter */
+  @Action
+  addnistIdFilter(cciId: string | string[]) {
+    this.context.commit('ADD_CCI', cciId);
+  }
+
+  @Mutation
+  ADD_CCI(cciId: string | string[]) {
+    this.nistIdFilter = this.nistIdFilter.concat(cciId);
+  }
+
+  /** Sets the CCI IDs filter */
+  @Mutation
+  SET_CCI(cciIds: string[]) {
+    this.nistIdFilter = cciIds;
+  }
+
+  /** Clears all CCI ID filters */
+  @Mutation
+  CLEAR_CCI() {
+    this.nistIdFilter = [];
+  }
+
+  // Description Filtering
+
+  /** Adds or replaces description filter in the search bar */
+  @Action
+  addDescriptionSearch(description: string) {
+    if (this.searchTerm.trim() !== '') {
+      this.context.commit(
+        'SET_SEARCH',
+        `${this.searchTerm} AND description:${description}`
+      );
+    } else {
+      this.context.commit('SET_SEARCH', `description:${description}`);
+    }
+  }
+
+  /** Adds description to filter */
+  @Action
+  addDescriptionFilter(description: string | string[]) {
+    this.context.commit('ADD_DESCRIPTION', description);
+  }
+
+  /** Adds a description filter */
+  @Mutation
+  ADD_DESCRIPTION(description: string | string[]) {
+    this.descriptionSearchTerms =
+      this.descriptionSearchTerms.concat(description);
+  }
+
+  /** Sets the descriptions filter */
+  @Mutation
+  SET_DESCRIPTION(descriptions: string[]) {
+    this.descriptionSearchTerms = descriptions;
+  }
+
+  /** Clears all description filters */
+  @Mutation
+  CLEAR_DESCRIPTION() {
+    this.descriptionSearchTerms = [];
+  }
+
+  // Code filtering
+
+  /** Adds or replaces code filter in the search bar */
+  @Action
+  addCodeSearch(code: string) {
+    if (this.searchTerm.trim() !== '') {
+      this.context.commit('SET_SEARCH', `${this.searchTerm} AND code:${code}`);
+    } else {
+      this.context.commit('SET_SEARCH', `code:${code}`);
+    }
+  }
```

Comment:
A lot of these filters could generically be parameterized if the options to these functions were changed to be

```options: {field: string, value: string}```